interface/forms/group_attendance/save.php

```diff
@@ -0,0 +1,109 @@
+    $sql_for_table_ftga = "INSERT INTO form_therapy_groups_attendance (id, date, group_id, user, groupname, authorized, encounter_id, activity) " .
+        "VALUES(?,NOW(),?,?,?,?,?,?);";
+    $sqlBindArray = array();
+    array_push($sqlBindArray, $newid, $therapy_group, $_SESSION["authUser"], $_SESSION["authProvider"], $userauthorized, $encounter, '1');
+    sqlInsert($sql_for_table_ftga, $sqlBindArray);
+
+    $patientData = $_POST['patientData'];
+    foreach ($patientData as $pid => $patient){
+
+        //Insert into therapy_groups_participants_attendance table
+        insert_into_tgpa_table($newid, $pid, $patient);
+
+        //Check if to create appt and encounter for each patient (if has certain status and 'bottom' submit was pressed, not 'add_patient' submit).
+        $create_for_patient = if_to_create_for_patient($patient['status']);
+        if($create_for_patient && empty($_POST['submit_new_patient'])){
+
+            //Create appt for each patient (if there is appointment connected to encounter)
+            if(!empty($appt_data)){
+                insert_patient_appt($pid, $therapy_group, $appt_data['pc_aid'], $appt_data['pc_eventDate'], $appt_data['pc_startTime'], $patient);
+            }
+
+            //Create encounter for each patient
+            insert_patient_encounter($pid, $therapy_group, $group_encounter_data['date'], $patient, $appt_data['pc_aid']);
+
+        }
+
+    }
+
+    //If adding a new participant
+    if(isset($_POST['submit_new_patient'])){
+        $new_participant_id = $_POST['new_id'];
+        $new_comment = $_POST['new_comment'];
+        insert_new_participant($newid, $new_participant_id, $new_comment);
+        jumpToEdit($newid);
+    }
+    else
+        formJump();
+
+
+}
+//If editing a form
+elseif ($_GET['mode'] == 'update'){
+
+    //Update form_therapy_groups_attendance table
+    $id = $_GET['id'];
+    $sql_for_form_tga = "UPDATE form_therapy_groups_attendance SET date = NOW(), user = ?, groupname = ?, authorized = ? WHERE id = ?;";
+    $sqlBindArray = array();
+    array_push($sqlBindArray,  $_SESSION["authUser"], $_SESSION["authProvider"], $userauthorized, $id);
+    sqlInsert($sql_for_form_tga, $sqlBindArray);
+
+    //Delete from therapy_groups_participant_attendance table
+    $sql_delete_from_table_tgpa = "DELETE FROM therapy_groups_participant_attendance WHERE form_id = ?;";
+    sqlStatement($sql_delete_from_table_tgpa, array($id));
+
+    $patientData = $_POST['patientData'];
+    foreach ($patientData as $pid => $patient){
+
+        //Insert into therapy_groups_participants_attendance table
+        insert_into_tgpa_table($id, $pid, $patient);
+
+        //Check if to create appt and encounter for each patient (if has certain status and 'bottom' submit was pressed, not 'add_patient' submit).
+        $create_for_patient = if_to_create_for_patient($patient['status']);
+        if($create_for_patient && empty($_POST['submit_new_patient'])){
```

Comment:
We aren't sure how to use this function and what it does in the system, as we've never implemented it. Would you be able to elaborate more on the tracker functionality (what it does, where we can see it in action in the system, etc.)?  We will happily embed it in the code.
thanks,
shachar